src/mds/MDSRank.cc

```diff
@@ -556,6 +556,7 @@ MDSRank::~MDSRank()
 {
   if (hb) {
     g_ceph_context->get_heartbeat_map()->remove_worker(hb);
+    hb = NULL;
```

Comment:
```suggestion
    hb = nullptr;
```

would be idiomatic c++

Reply:
... and there is another `hb = NULL` in `shutdown()`.

Also, I think by the time we hit the destructor `hb` is already nullptr, but its better to be safe.

Reply:
> would be idiomatic c++
Done.

Reply:
> ... and there is another `hb = NULL` in `shutdown()`.
> 
> Also, I think by the time we hit the destructor `hb` is already nullptr, but its better to be safe.

Yeah, agree. I am not sure whether will there have some cases that the `mds_rank` hasn't initialized yet and directly call the `~MDSRank()`.

Reply:
hmmm..

Let's change to nullptr. This looks fine!